rust/lance-linalg/src/distance/l2.rs

```diff
@@ -67,10 +67,13 @@ where
 ///
 /// This is pub for test/benchmark only. use [l2] instead.
 #[inline]
-pub fn l2_scalar<T: Float + Sum + AddAssign + AsPrimitive<f32>, const LANES: usize>(
+pub fn l2_scalar<T, Output: Float + Sum + AddAssign + 'static, const LANES: usize>(
     from: &[T],
     to: &[T],
-) -> f32 {
+) -> Output
+where
+    T: AsPrimitive<Output>,
```

Comment:
Any particular reason to move the `AsPrimitive` bound here?  Or just trying to declutter things?

Reply:
I actually thought it wasn't possible, but turns out it is.